app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -166,7 +166,7 @@ class AppsPreferences(private val prefs: SharedPreferences) {
 
         return when {
             pref.toLowerCase() == "ssh" || (app.supportsCli && !app.supportsGui) -> SshTypePreference
-            pref.toLowerCase() == "vnc" || (!app.supportsCli && app.supportsGui) -> VncTypePreference
+            pref.toLowerCase() == "vnc" -> VncTypePreference
```

Comment:
I believe the `||` branch in the `vnc` conditional expression is still necessary to get the defaulting behavior we want.

Reply:
Yeah!!! This Update Will Be Awesome When It's Done. I Can't Wait

Reply:
What do you think about this ordering of conditionals?

```
return when {
    pref.toLowerCase() == "ssh" || (app.supportsCli && !app.supportsGui) -> SshTypePreference
    pref.toLowerCase() == "xsdl" -> XsdlTypePreference
    pref.toLowerCase() == "vnc" || (!app.supportsCli && app.supportsGui) -> VncTypePreference
    else -> PreferenceHasNotBeenSelected
}
```

If the app already has XSDL set as the client type, the [appDetail] page will reflect that.  Otherwise if it's not set to SSH or VNC, then it will default to VNC if the app doesn't support CLI but supports GUI.